### ISSUE TITLE
Document how to run hitch on Travis-CI and other CI services

### ISSUE THREAD

**Issue by thedrow** [hitchtest/hitch#15]
Since this is a testing framework, explaining how to run hitch on CI services is crucial.

**Comment by crdoconnor** [comment]
Yeah, I need to add a page on readthedocs about that.

There's is an example travis config already that has been running for some time: https://github.com/hitchtest/django-remindme/blob/master/.travis.yml

If you write your own tests on your own project, there's a good chance that one will suffice for you. It's not that complicated and the set up steps are much the same as they are on your development environment.

I've also got an example settings file that is suitable for use on a CI environment on the example project: https://github.com/hitchtest/django-remindme-tests/blob/master/ci.settings 

That switches off fail fast and runs firefox in headless mode.

**Comment by crdoconnor** [comment]
Added!

https://hitchtest.readthedocs.org/en/latest/howto/continuous_integration.html

Any comments?

**Comment by thedrow** [comment]
Yes, document usage with tox and if there are any folders to be cached in order to get faster build time on Travis.
You can also recommend to create a docker container which already has everything set up for hitch to run tests quickly.

**Comment by crdoconnor** [comment]
Tox shouldn't be used with hitch. Hitch already does everything tox does anyway (except run your existing unit tests, but even that's coming very soon).

Travis, annoyingly, won't let you apt-get install on their container based infrastructure and their workaround is just awful. Not workable. It'll remain slow for now.

http://docs.travis-ci.com/user/migrating-from-legacy/#What-are-the-restrictions%3F

If you've got a private repository, you can cache the ~/.hitchpkg folder, though. I'll add that to the docs.

Docker can be used to run hitch, but since A) it's already a one step process to get it up and running and B) docker doesn't work on Mac and C) getting firefox to run non-headless in a container seems to be monstrously hard I'm not sure I see the point.

**Comment by thedrow** [comment]
Their new KVM infrastructure allows you to run Docker containers. See http://docs.travis-ci.com/user/docker/
Caching also works for open source now.

**Comment by crdoconnor** [comment]
Huh. I guess the other doc page is out of date.

Let me take another look.
